File: AndroidStudio/Individual9/app/src/main/java/com/example/individual9/MainActivity.java
package com.example.individual9;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView image1 = findViewById(R.id.image1);
        ImageView image2 = findViewById(R.id.image2);
        ImageView image3 = findViewById(R.id.image3);
        ImageView image4 = findViewById(R.id.image4);
        Button buttonCloseApp = findViewById(R.id.buttonCloseApp);

        image1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openSecondActivity(R.drawable.gato1);
            }
        });

        image2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openSecondActivity(R.drawable.gato2);
            }
        });

        image3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openSecondActivity(R.drawable.gato3);
            }
        });

        image4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openSecondActivity(R.drawable.gato4);
            }
        });

        buttonCloseApp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    private void openSecondActivity(int imageResource) {
        Intent intent = new Intent(MainActivity.this, SecondActivity.class);
        intent.putExtra("imageResource", imageResource);
        startActivity(intent);
    }
}

File: AndroidStudio/SuperHeroList/app/src/main/java/com/example/superherolist/MainActivity.kt
package com.example.superherolist

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private val superheroList = mutableListOf<Superhero>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val resultTextView = findViewById<TextView>(R.id.resultTextView)
        val addHeroButton = findViewById<Button>(R.id.btnAddHero)


        addHeroButton.setOnClickListener {
            addSuperheroes()
            resultTextView.text = superheroList.joinToString(separator = "\n") { hero ->
                "${hero.superhero} - ${hero.publisher} - ${hero.realName} - ${hero.photo}"
            }
        }
    }

    private fun addSuperheroes() {
        superheroList.add(Superhero("Spider-Man", "Marvel", "Peter Parker", "spiderman_photo.jpg"))
        superheroList.add(Superhero("Batman", "DC", "Bruce Wayne", "batman_photo.jpg"))
        superheroList.add(Superhero("Superman", "DC", "Clark Kent", "superman_photo.jpg"))
    }
}